components/net/resource_thread.rs

```diff
@@ -429,10 +431,116 @@ pub struct CoreResourceManager {
     devtools_chan: Option<Sender<DevtoolsControlMsg>>,
     swmanager_chan: Option<IpcSender<CustomResponseMediator>>,
     filemanager: FileManager,
-    fetch_pool: rayon::ThreadPool,
+    thread_pool: Arc<CoreResourceThreadPool>,
     certificate_path: Option<String>,
 }
 
+/// The state of the thread-pool used by CoreResource.
+enum ThreadPoolState {
+    /// Pool is inactive, cannot spawn additional work,
```

Comment:
Should this instead be a struct with `active_threads: u32, is_active: bool`?

Reply:
I personally liked the enum, so it's your call...

Reply:
I'm not a big fan of enums where both variants contain the same payload meaning the same thing.

Reply:
Ok, I will change it then.